### PR TITLE
Feature/test batery

### PR DESCRIPTION
**Pull request by xkevin190**
### Contribution description

the module was added and tested the library to do unit tests of riot, for the first tests all the storage module of the firmware was tested.

### Testing procedure

cd test/storage

make BOARD="your_board" flash test

the module was added and tested the library to do unit tests of riot, for the first tests all the storage module of the firmware was tested.

### CODE REVIEW

File: tests/Makefile.tests_common [hunk 1]
```diff
@@ -13,8 +13,10 @@ ifneq (,$(filter tests_driver_%,$(APPLICATION)))
...
 
 BOARD ?= m4a-mb
-RIOTBASE ?= $(CURDIR)/../RIOT
 QUIET ?= 1
+RIOTBASE ?= ../../RIOT
+EXTERNAL_BOARD_DIRS ?= ../../boards
+EXTERNAL_MODULE_DIRS += ../../firmware/sys
 
 # DEVELHELP enabled by default for all tests, set 0 to disable
 DEVELHELP ?= 1
```
Comment (eduazocar) [inline tests/Makefile.tests_common:22]: Add EOF

File: tests/storage/tests/01-run.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# Copyright (C) 2017 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from testrunner import run_check_unittests
+
```
Comment (eduazocar) [inline tests/storage/tests/01-run.py:11]: I know that it is python, but you have an empty line

**Review by eduazocar — COMMENTED** [review]
Everything looks good for me, only make a little changes :+1:

**Review by luisan00 — CHANGES_REQUESTED** [review]
1. static test errors should be fixed before to continue.
2. error compiling for `m4a-24g` and possibly `m4a-mb` (not tested)

**Comment by luisan00** [comment]
test command **is**: 
```make BOARD="your_board" flash test```

File: tests/storage/main.c [hunk 3]
```diff
@@ -26,7 +26,6 @@
```
Comment (RocioRojas) [inline tests/storage/main.c:26]: remove blank line

File: tests/Makefile.tests_common [hunk 4]
```diff
@@ -14,9 +14,9 @@ endif
...
 QUIET ?= 1
-RIOTBASE ?= ../../RIOT
-EXTERNAL_BOARD_DIRS ?= ../../boards
-EXTERNAL_MODULE_DIRS += ../../firmware/sys
+RIOTBASE ?= $(CURDIR)/../../RIOT
+EXTERNAL_BOARD_DIRS ?=  $(CURDIR)/../../boards
+EXTERNAL_MODULE_DIRS += $(CURDIR)/../../firmware/sys
 
 # DEVELHELP enabled by default for all tests, set 0 to disable
 DEVELHELP ?= 1
```
Comment (RocioRojas) [inline tests/Makefile.tests_common:22]: add EOF

**Review by RocioRojas — APPROVED** [review]

**Comment by luisan00** [comment]
Squash it please

**Comment by luisan00** [comment]
Looks good, wait for static test and ready

**Review by luisan00 — APPROVED** [review]
Ready to go ahead

**Pull request merged**